### PR TITLE
feat(ibm): add ability to get image id from name

### PR DESCRIPTION
**Pull request by a-dubs**
This change makes it much easier to select an image to launch an instance from.

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-1!6.2.1
+1!6.2.3
```
Comment (uhryniuk) [inline VERSION:1]: Should patch jump from x.x.1 -> x.x.3?

Comment (a-dubs) [reply VERSION:1]: uhhhh nope. you saw nothing. this never happened.

**Review by uhryniuk — APPROVED** [review]
Looks good, sire

File: pycloudlib/ibm/cloud.py [hunk 2]
```diff
@@ -403,6 +403,17 @@ def _get_or_create_key(self) -> str:
         self.created_keys.append(key_id)
         return key_id
 
+    def get_image_id_by_name(self, name: str) -> str:
```
Comment (holmanb) [inline pycloudlib/ibm/cloud.py:406]: Nice! What sorts of images are you testing? Think that this is something we might want for other clouds? I can imagine this being useful, for example, to test cloud-init for non-ubuntu clouds.

Maybe we should put an abstract method in the base class as well so we can have a standardized interface for other clouds to match?

Comment (a-dubs) [reply pycloudlib/ibm/cloud.py:406]: let me take a look at that! would definitely be something that would be both easy to add to all clouds and also just generally useful.

**Review by holmanb — APPROVED** [review]
LGMT!

Once the flake8 failures are fixed, I'll merge.

**Review by holmanb — APPROVED** [review]

**Pull request merged**